client/src/app.jsx

```diff
@@ -48,17 +51,17 @@ class App extends React.Component {
     })
   }
 
-  getStateData() {
+  getStateData(product_id) {
```

Comment:
Passing in a param here definitely makes sense. It does a lot to clean up the promise.All